cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -305,6 +309,44 @@ class device_row_comparator {
   {
   }
 
+  /**
+   * @brief Construct a function object for performing a lexicographic
+   * comparison between the rows of two tables.
+   * This is a special overload to allow device-side construction of the
+   * comparator for cases where no preprocessing is needed, i.e. tables with
+   * non-nested type columns.
+   *
+   * @param check_nulls Indicates if any input column contains nulls.
+   * @param lhs The first table
+   * @param rhs The second table (may be the same table as `lhs`)
+   * @param column_order Optional, device array the same length as a row that indicates the desired
+   * ascending/descending order of each column in a row. If `nullopt`, it is assumed all columns are
+   * sorted in ascending order.
+   * @param null_precedence Optional, device array the same length as a row and indicates how null
+   * values compare to all other for every column. If `nullopt`, then null precedence would be
+   * `null_order::BEFORE` for all columns.
+   * @param comparator Physical element relational comparison functor.
+   */
+  template <bool nested_disable = not has_nested_columns, CUDF_ENABLE_IF(nested_disable)>
```

Comment:
Interesting phrasing. I'll remember this for when I need it.

Is `nested_disable` used elsewhere?
If not, would `CUDF_ENABLE_IF(not has_nested_columns)` be less readable?

Reply:
I can change the variable name if you'd like, but I cannot use the class-template `has_nested_columns` directly here.

Let me explain why, and my apologies if you already knew this but I misunderstood what you are suggesting.

If I did:
```
template <CUDF_ENABLE_IF(not has_nested_columns)`
```
then we'd not be able to trigger SFINAE. This is because `has_nested_columns` is a class-template. So, when you write `device_comparator<true> d_comp`, the substitution for the enable-if on the constructor will trigger a failure as `enable_if::type` will be ill-formed.

As a result of which, we want to defer the SFINAE at the member function level so that it behaves the way it is intended to: as an overload.